yt_dlp/extractor/tenplay.py

```diff
@@ -101,30 +108,145 @@ class TenPlayIE(InfoExtractor):
+            raise
+
     def _real_extract(self, url):
         content_id = self._match_id(url)
         data = self._download_json(
-            'https://10.com.au/api/v1/videos/' + content_id, content_id)
-
-        video_data = self._download_json(
-            f'https://vod.ten.com.au/api/videos/bcquery?command=find_videos_by_id&video_id={data["altId"]}',
-            content_id, 'Downloading video JSON')
-        # Dash URL 404s, changing the m3u8 format works
-        m3u8_url = self._request_webpage(
-            HEADRequest(update_url_query(video_data['items'][0]['dashManifestUrl'], {
-                'manifest': 'm3u',
-            })),
-            content_id, 'Checking stream URL').url
-        if '10play-not-in-oz' in m3u8_url:
-            self.raise_geo_restricted(countries=['AU'])
-        if '10play_unsupported' in m3u8_url:
-            raise ExtractorError('Unable to extract stream')
-        # Attempt to get a higher quality stream
-        formats = self._extract_m3u8_formats(
-            m3u8_url.replace(',150,75,55,0000', ',500,300,150,75,55,0000'),
-            content_id, 'mp4', fatal=False)
-        if not formats:
-            formats = self._extract_m3u8_formats(m3u8_url, content_id, 'mp4')
+            f'https://10.com.au/api/v1/videos/{content_id}', content_id)
+
+        video_data, urlh = self._call_playback_api(content_id)
+        content_source_id = video_data['dai']['contentSourceId']
+        video_id = video_data['dai']['videoId']
+        auth_token = urlh.get_header('x-dai-auth')
+        if not auth_token:
+            raise ExtractorError('Failed to get DAI auth token')
+
+        dai_data = self._download_json(
+            f'https://pubads.g.doubleclick.net/ondemand/hls/content/{content_source_id}/vid/{video_id}/streams',
+            content_id, note='Downloading DAI JSON',
+            data=urlencode_postdata({'auth-token': auth_token}))
+
+        # Ignore subs to avoid ad break cleanup
+        formats, _ = self._extract_m3u8_formats_and_subtitles(
+            dai_data['stream_manifest'], content_id, 'mp4')
+
+        already_have_1080p = False
+        for fmt in formats:
+            m3u8_doc = self._download_webpage(
+                fmt['url'], content_id, note='Downloading m3u8 information')
+            m3u8_doc = self._filter_ads_from_m3u8(m3u8_doc)
+            fmt['hls_media_playlist_data'] = m3u8_doc
+            if fmt.get('height') == 1080:
+                already_have_1080p = True
+
+        # Attempt format upgrade
+        if not already_have_1080p and m3u8_doc and re.search(r'(?m)-(?:300|150|75|55)0000-\d+\.ts$', m3u8_doc):
+            m3u8_doc = re.sub(r'(?m)-(?:300|150|75|55)0000-(\d+)\.ts$', r'-5000000-\1.ts', m3u8_doc)
+            m3u8_doc = re.sub(r'-(?:300|150|75|55)0000\.key"', r'-5000000.key"', m3u8_doc)
+            formats.append({
+                'format_id': 'upgrade-attempt-1080p',
```

Comment:
I think there's supposed to be an upgrade-attempt-720p as well. I'll find an example URL soon™️